py4DSTEM/utils/configuration_checker.py

```diff
@@ -1,61 +1,93 @@
 #### this file contains a function/s that will check if various
 # libaries/compute options are available
 import importlib
-from operator import mod
-
-# list of modules we expect/may expect to be installed
-#  as part of a standard py4DSTEM installation
-# this needs to be the import name e.g. import mp_api not mp-api
-modules = [
-    "crystal4D",
-    "cupy",
-    "dask",
-    "dill",
-    "distributed",
-    "gdown",
-    "h5py",
-    "ipyparallel",
-    "jax",
-    "matplotlib",
-    "mp_api",
-    "ncempy",
-    "numba",
-    "numpy",
-    "pymatgen",
-    "skimage",
-    "sklearn",
-    "scipy",
-    "tensorflow",
-    "tensorflow-addons",
-    "tqdm",
-]
-
-# currently this was copy and pasted from setup.py,
-# hopefully there's a programatic way to do this.
-module_depenencies = {
-    "base": [
-        "numpy",
-        "scipy",
-        "h5py",
-        "ncempy",
-        "matplotlib",
-        "skimage",
-        "sklearn",
-        "tqdm",
-        "dill",
-        "gdown",
-        "dask",
-        "distributed",
-    ],
-    "ipyparallel": ["ipyparallel", "dill"],
-    "cuda": ["cupy"],
-    "acom": ["pymatgen", "mp_api"],
-    "aiml": ["tensorflow", "tensorflow-addons", "crystal4D"],
-    "aiml-cuda": ["tensorflow", "tensorflow-addons", "crystal4D", "cupy"],
-    "numba": ["numba"],
+from importlib.metadata import requires
+import re
+
+# need a mapping of pypi/conda names to import names
+import_mapping_dict = {
+    "scikit-image": "skimage",
+    "scikit-learn": "sklearn",
+    "scikit-optimize": "skopt",
+    "mp-api": "mp_api",
 }
 
 
+# programatically get all possible requirements in the import name style
+def get_modules_list():
+    # Get the dependencies from the installed distribution
+    dependencies = requires("py4DSTEM")
+
+    # Define a regular expression pattern for splitting on '>', '>=', '='
+    delimiter_pattern = re.compile(r">=|>|==|<|<=")
+
+    # Extract only the module names without versions
+    module_names = [
+        delimiter_pattern.split(dependency.split(";")[0], 1)[0].strip()
+        for dependency in dependencies
+    ]
+
+    # translate pypi names to import names e.g. scikit-image->skimage, mp-api->mp_api
+    for index, module in enumerate(module_names):
+        if module in import_mapping_dict.keys():
+            module_names[index] = import_mapping_dict[module]
+
+    return module_names
+
+
+# programatically get all possible requirements in the import name style,
+# split into a dict where optional import names are keys
+def get_modules_dict():
+    package_name = "py4DSTEM"
+    # Get the dependencies from the installed distribution
+    dependencies = requires(package_name)
+
+    # set the dictionary for modules and packages to go into
+    modules_dict = {
+        "base": [],
+        "acom": [],
+        "aiml": [],
+        "aiml-cuda": [],
+        "cuda": [],
+        "numba": [],
```

Comment:
Is there a way to generate this programatically? We may add or modify the list of optional install pathways in the future, and this would have to change to match.

Reply:
updated, and tested by adding `"test": ["pytest", "flake8", "black"]`  to the setup.py which populates correctly